core/common/src/main/java/alluxio/util/network/NettyUtils.java

```diff
@@ -162,18 +139,26 @@ public static boolean isDomainSocketSupported(WorkerNetAddress workerNetAddress)
 
   /**
    * Note: Packet streaming requires {@link io.netty.channel.epoll.EpollMode} to be set to
-   * LEVEL_TRIGGERED which is not supported in netty versions < 4.0.26.Final. Without shading
-   * netty in Alluxio, we cannot use epoll.
+   * LEVEL_TRIGGERED which is not supported in netty versions < 4.0.26.Final. Without shading netty
+   * in Alluxio, we cannot use epoll.
    *
-   * @return {@link ChannelType} to use
+   * @param key the property key for looking up the configured channel type
+   * @return the channel type to use
    */
-  private static ChannelType getChannelType() {
-    try {
-      EpollChannelOption.class.getField("EPOLL_MODE");
-    } catch (Throwable e) {
-      LOG.warn("EPOLL_MODE is not supported in netty with version < 4.0.26.Final.");
-      return ChannelType.NIO;
+  private static ChannelType getChannelType(PropertyKey key) {
+    ChannelType configured = Configuration.getEnum(key, ChannelType.class);
+    if (configured == ChannelType.EPOLL) {
+      if (!Epoll.isAvailable()) {
+        LOG.info("EPOLL is not available, will use NIO");
+        return ChannelType.NIO;
+      }
+      try {
+        EpollChannelOption.class.getField("EPOLL_MODE");
+      } catch (Throwable e) {
+        LOG.warn("EPOLL_MODE is not supported in netty with version < 4.0.26.Final, will use NIO");
```

Comment:
Not sure - @calvinjia do you know the history of this check?

Reply:
Yeah, if for some reason we are using an older version of netty, we will avoid using epoll. We need to do this check because epoll will be available, but not level triggered mode.

Reply:
ack